### PR TITLE
MAISTRA-2051 fixes for maistra install

### PR DESCRIPTION
**Pull request by rcernich**
Signed-off-by: rcernich <rcernich@redhat.com>

Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by rcernich** [comment]
@bison, @dgn FYI.  These are the changes I've had to make so far to get pilot working with maistra.  For whatever reason, it doesn't appear that endpoint/pod watch is working for namespaces outside the system namespace (e.g. istio-system).

File: pkg/kube/client.go [hunk 1]
```diff
@@ -461,6 +461,7 @@ func (c *client) RunAndWait(stop <-chan struct{}) {
 	c.metadataInformer.Start(stop)
 	c.istioInformer.Start(stop)
 	c.serviceapisInformers.Start(stop)
+	c.memberRoll.Start(stop)
```
Comment (rcernich) [inline pkg/kube/client.go]: I'm not sure if this is correct, but it seemed the member roll controller needed to be started somewhere.

Comment (bison) [reply pkg/kube/client.go]: Oops. Yeah, but we need to check if it is `nil` before starting it.

File: pkg/kube/inject/inject.go [hunk 2]
```diff
@@ -1100,6 +1100,28 @@ func potentialPodName(metadata *metav1.ObjectMeta) string {
 	return ""
 }
 
+
+// rewriteCniPodSpec will check if values from the sidecar injector Helm
+// values need to be inserted as Pod annotations so the CNI will apply
+// the proper redirection rules.
+func rewriteCniPodSpec(annotations map[string]string, spec *SidecarInjectionSpec) {
```
Comment (bison) [inline pkg/kube/inject/inject.go:1106]: Oops. I had purposefully skipped this stuff, cause I think it's fixed upstream, but part of it may have only made it into the 1.9 release.

File: pkg/kube/inject/webhook.go [hunk 3]
```diff
@@ -714,6 +714,10 @@ func injectPod(req InjectionParameters, partialInjection bool) ([]byte, error) {
 		annotations[k] = v
 	}
 
+	if len(spec.PodRedirectAnnot) != 0 {
```
Comment (jwendell) [inline pkg/kube/inject/webhook.go]: nit: this `if` is redundant with the one inside the function

File: tools/istio-iptables/pkg/cmd/root.go [hunk 4]
```diff
@@ -192,6 +194,16 @@ func init() {
 	}
 	viper.SetDefault(constants.ProxyGID, "")
 
+	rootCmd.Flags().Bool(constants.RedirectDNS, false,
```
Comment (dgn) [inline tools/istio-iptables/pkg/cmd/root.go:197]: why do we need this redirectDNS option? I think if we want to change the behaviour around DNS interception we should do it in a separate PR, it shouldn't be required to get an install to work?

Comment (rcernich) [reply tools/istio-iptables/pkg/cmd/root.go:197]: This isn't a change.  The logic for enabling this used to use an environment variable to enable/disable this (`ISTIO_META_DNS_CAPTURE`), but that doesn't work for CNI, so I added a command line option.  Viper should be initializing this option using the value from the env, if specified, when running in the init container.  This is basically synchronizing the existing iptables capabilities used by init container with cni.

Comment (rcernich) [reply tools/istio-iptables/pkg/cmd/root.go:197]: fwiw, this change is encapsulated in a separate commit.

**Review by dgn — APPROVED** [review]
